docs/deepdive.md

```diff
@@ -0,0 +1,126 @@
+# Ollama Deep Dive
```

Comment:
I like some of the content here but after reviewing, most of it belongs in the FAQ to help specific questions (vs a broader deep dive topic)